flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShim.java

```diff
@@ -232,4 +233,10 @@ SimpleGenericUDAFParameterInfo createUDAFParameterInfo(ObjectInspector[] params,
 	 * Converts a hive date instance to LocalDate which is expected by DataFormatConverter.
 	 */
 	LocalDate toFlinkDate(Object hiveDate);
+
+	/**
+	 * Converts a Hive primitive java object to corresponding Writable object. Throws CatalogException if the conversion
+	 * is not supported.
+	 */
+	Writable hivePrimitiveToWritable(Object value) throws CatalogException;
```

Comment:
Throwing a catalog ex doesn't seem very intuitive, even though CatalogEx is also a runtime exception. Maybe just remove CatalogEx from the signature and throw a runtime exception when problem occurs.

Reply:
I'll change to throw `FlinkHiveException`. It's also a runtime exception but more specific to Hive connector.